### PR TITLE
Scaffold books api with basic, stubbed get endpoint

### PR DESCRIPTION
**Pull request by kingscott**
## Description

Using some of the logic from a go-kit example: https://github.com/go-kit/examples/tree/master/profilesvc, scaffold the KR books API. Right now it implements just one service, `books`. 

I also added a small start script to make starting the API easier.

### CODE REVIEW

**Comment by kingscott** [comment]
If you want me to add some more comments @erullmann I can

File: README.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+```go
+go run ./cmd/main.go
+```
+
+The API should be running on port `8080`.
+
+Or use the start script:
+
+```bash
+chmod +x ./start.sh
```
Comment (erullmann) [inline README.md:18]: If you chmod it before committing it, won't it be marked as an executable?

File: cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
```
Comment (erullmann) [inline cmd/main.go:5]: Does the indentation here mean something?

File: pkg/v1/endpoints/books/request.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package books
+
+import (
+    "github.com/King-Rullmann-Book-Club/books-api/pkg/v1/service/books"
```
Comment (erullmann) [inline pkg/v1/endpoints/books/request.go:4]: Had a general question about this type of import. Does the repo need to be public for this to work? How does uncommitted/unmerged code (like here where the books service hasn't been made yet) get used? Why is this preferable to a relative path like (`import "books-api/pkg/v1/service/books"`)? Can non-github sources be used (e.g. gitlab).

File: pkg/v1/service/books/service.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package books
+
+import "context"
+
+type Service interface {
+    GetBook(ctx context.Context, id string) (Book, error)
+}
+
+type Book struct {
+    Name string
```
Comment (erullmann) [inline pkg/v1/service/books/service.go:10]: nit: Use `Title`

Comment (erullmann) [reply pkg/v1/service/books/service.go:10]:
```suggestion
    Title string
```

File: cmd/api/client.go [hunk 5]
```diff
@@ -0,0 +1,3 @@
+package api
+
+// TODO
```
Comment (erullmann) [inline cmd/api/client.go:3]: First of many I'm sure. :P

File: pkg/v1/service/books/books.go [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package books
+
+import (
+    "context"
+)
+
+// GetBook returns the details of a given book ID. 
+func (s *bookSvc) GetBook(ctx context.Context, id string) (Book, error) {
+    return Book{Name: "Perdido Street Station"}, nil
```
Comment (erullmann) [inline pkg/v1/service/books/books.go:9]: Did you have thoughts on a data store? SQLite for example?

File: pkg/v1/transport/books/transport.go [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        return nil, errors.New("Unable to get parameter, bad route")
+    }
+    return ep.GetBookRequest{ID: id}, nil
+}
+
+// errorer is implemented by all concrete response types that may contain
+// errors. It allows us to change the HTTP response code without needing to
+// trigger an endpoint (transport-level) error. For more information, read the
+// big comment in endpoints.go.
+type errorer interface {
```
Comment (erullmann) [inline pkg/v1/transport/books/transport.go:50]: I think the big comment in `endpoints.go` is missing. But I guess how I read this is that business logic errors are encoded in the interface (so like a not found, etc), but a more technical/transport level exception, such as a resource not being available, bad argument, etc would be returned through the `error` of the `interface{}, error` pair?

**Review by erullmann — APPROVED** [review]
LG, I have a few questions about various details.

File: cmd/main.go (hunk 2 above)
Comment (erullmann) [reply cmd/main.go:5]: I mean that its different from the other lines.

File: README.md (hunk 1 above)
Comment (kingscott) [reply README.md]: I'm not too sure. I just make sure to include in case we forget :D

File: cmd/main.go (hunk 2 above)
Comment (kingscott) [reply cmd/main.go]: Nope, just editor things.

File: pkg/v1/endpoints/books/request.go (hunk 3 above)
Comment (kingscott) [reply pkg/v1/endpoints/books/request.go]: Not necessarily, this is just standard [go modules](https://go.dev/ref/mod#modules-overview). I've never tested out relative imports.

File: pkg/v1/service/books/service.go (hunk 4 above)
Comment (kingscott) [reply pkg/v1/service/books/service.go]: Will fix this.

File: pkg/v1/service/books/books.go (hunk 6 above)
Comment (kingscott) [reply pkg/v1/service/books/books.go]: Can we deploy with SQLite? I was thinking MySQL or Postgresql.

File: pkg/v1/transport/books/transport.go (hunk 7 above)
Comment (kingscott) [reply pkg/v1/transport/books/transport.go]: It's missing a general comment about how the content of `endpoints.go` work? 

I believe your read is correct. I'm new to this style of error handling. Normally, I'd have an encoded response for each endpoint. I borrowed this from go-kit examples, as it is a reusable method, similar to what we discussed the other day. The idea is to make sure to return an error string, not an interface.

File: pkg/v1/service/books/books.go (hunk 6 above)
Comment (erullmann) [reply pkg/v1/service/books/books.go]: For two people? Yeah SQLite works.

File: pkg/v1/transport/books/transport.go (hunk 7 above)
Comment (erullmann) [reply pkg/v1/transport/books/transport.go]: I mean the line `For more information, read the big comment in endpoints.go.` There isn't really a comment in `endpoints.go` that fits. Unless I'm missing something.

Comment (kingscott) [reply pkg/v1/transport/books/transport.go]: Ahh, I see. I can add that comment to `endpoints.go`.

**Pull request closed without merging**